logicalRequirements.md

```diff
@@ -285,6 +285,33 @@ __Additional considerations__
 
 Please note that fulfilling this logical element requires interaction with the door in the adjacent room to be possible (so no active locks on it, and fulfilling its interaction requirements). Fulfilling this logical element also causes the room to be reset, which means all obstacles respawn.
 
+#### adjacentJumpway object
+An `adjacentJumpway` object represents the need for Samus to be able to jump into the room from a door frame or platform in an adjacent room. It has the following properties:
+* _fromNode:_ Indicates from what door this logical requirement expects Samus to enter the room
+* _type:_ Possible values are "doorFrameBelow" and "platformBelow". The logical requirement can only be satisfied by jumpways having a matching type.
+* _minHeight:_ Minimum value of "height" on a jumpway to be able to satisfy this requirement. For a door frame jumpway, this expresses that the door frame extends at least a certain distance below the door transition. Likewise, for a platform  jumpway, it expresses that the platform must be positioned at least a certain distance below the door transition.
+* _maxHeight:_ Maximum value of "height" on a jumpway to be able to satisfy this requirement. For a platform jumpway, this expresses that the platform must be positioned at most a certain distance below the door transition.
+* _maxLeftPosition:_ Maximum value of "leftPosition" on a jumpway to be able to satisfy this requirement. This applies only to platform jumpways and expresses that the platform extends at least a certain distance to the left.
```

Comment:
Added some detail to clarify the meaning here. `jumpwayLength` in many cases can't encode the requirements because some strats require entering the door in a specific way (e.g. through a specific side), and whether this is possible or not can depend on the position where the jumpway ends, not only its length.

Reply:
Yes, jumpwayLength would need more information; jump point, and direction, and maybe something else.  jumpPoint could have some issues, but maybe it could be defined broad enough to work out.

LeftRight seems to have issues with broken up, small jumpways.  But once you're a little bit away from the door that problem goes away.  For long jumpways, and single tile jumpways its pretty good, so maybe I like it more.  But its hard to know if some of these jumpways are being captured or not without looking at the numbers.

Reply:
Not sure I'm seeing the problem with the "LeftRight" here. With broken up small jumpways we can just define multiple jumpways; this only happens in a couple rooms. With a "jump point" approach we would have to define multiple jump points, and there would need to be even more of them. In the "LeftRight" approach often a single jumpway can be used with a jump at multiple points without needing to define a new jumpway.

The idea here is for the jumpways to define which platforms/doorframes exist and the relevant information about their geometry, not to dictate how they are used (which is instead in the scope of the strat in the other room)

Reply:
I'm talking about `adjacentJumpway` in both cases above.

To use a 2 tile runway from the height 6 jumpways, if I understand it right, you would write 
Climb left = `maxLeft:-2.5, minRight:-0.5`
Climb right = `maxLeft:0.5, minRight:2.5`
Wasteland = `maxLeft:-1, minRight:1`
Crab Maze, WS shaft, WS basement would match any of the above.
And long jumpway in either direction would be fine.
It takes a need for a short runway, that is also below the door and I don't know that it's that problematic.

Building run speed then using spacejump to reach the door would also be individually added for each place, like Sporespawn kihunters.
And after writing my example out, using 0.5 in `adjacentJumpway` is pretty weird if you weren't targeting specific jumpways.

I'm naturally thinking in terms of general use jumpways, but that is harder to do.

Reply:
For an "adjacentJumpway" requiring only a single tile-platform at an arbitrary position below the door (or one tile to the side of the door), we can just use "maxLeftPosition": 2, "minRightPosition": -2. For example, Mt. Everest has strats like this. So there's no need to list out 3 different adjacentJumpways